pkg/kubernetes/node/v1alpha1/node_test.go

```diff
@@ -0,0 +1,179 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func fakeEmptyNodeList(noOfNodes int) *corev1.NodeList {
+	list := &corev1.NodeList{}
+	for i := 0; i < noOfNodes; i++ {
+		node := corev1.Node{}
+		list.Items = append(list.Items, node)
+	}
+	return list
+}
+
+func fakeAPINodeList(nodeNames []string) *corev1.NodeList {
+	if len(nodeNames) == 0 {
+		return nil
+	}
+	list := &corev1.NodeList{}
+	for _, name := range nodeNames {
+		node := corev1.Node{}
+		node.SetName(name)
+		list.Items = append(list.Items, node)
+	}
+	return list
+}
+
+func fakeAPINodeListFromNameStatusMap(nodes map[string]corev1.NodeConditionType) []*node {
+	nlist := []*node{}
+	for k := range nodes {
+		n := &corev1.Node{}
+		fakeNodeCondition := corev1.NodeCondition{
+			Reason: kubeletReady,
+			Type:   nodes[k],
+		}
+		n.SetName(k)
+		n.Status.Conditions = append(n.Status.Conditions, fakeNodeCondition)
+		nlist = append(nlist, &node{n})
+	}
+	return nlist
+}
+
+func TestListBuilderFuncWithAPIList(t *testing.T) {
+	tests := map[string]struct {
+		availableNodes  []string
+		expectedNodeLen int
```

Comment:
it may be `expectedNodeCount` len is very specific to array.

Reply:
Done